tests/test_runner.py

```diff
@@ -20,3 +26,54 @@ def test_docker_runner():
                 "--remove-orphans",
             ],
         )
+
+
+class MockKfpClient:
+    def __init__(self, host):
+        self.host = host
+        self._experiments = {"Default": SimpleNamespace(id="123")}
+
+    def get_experiment(self, experiment_name):
+        try:
+            return self._experiments[experiment_name]
+        except KeyError:
+            raise ValueError
+
+    def create_experiment(self, experiment_name):
+        self._experiments[experiment_name] = SimpleNamespace(id="456")
+        return self.get_experiment(experiment_name)
+
+    def run_pipeline(self, experiment_id, job_name, pipeline_package_path):
+        return SimpleNamespace(id="xyz")
+
+
+def test_kubeflow_runner():
+    input_spec_path = str(VALID_PIPELINE / "kubeflow_pipeline.yml")
+    with mock.patch(
+        "kfp.Client",
+        new=MockKfpClient,
+    ):
+        runner = KubeflowRunner(host="some_host")
+        runner.run(input_spec=input_spec_path)
```

Comment:
What are we testing here exactly? I know it's difficult since we're mocking the pipeline run method. Can we maybe validate the `run_url`?

Reply:
We are testing nothing explicit but if the runner would error then the test would fail. I'll add an assert to see if the url is passed properly.